jax/_src/lax/lax.py

```diff
@@ -2423,7 +2423,9 @@ def _convert_to_hlo_attr(self, lhs_dtype: DTypeLike,
 
 
 def dot(lhs: Array, rhs: Array, precision: PrecisionLike = None,
-        preferred_element_type: DTypeLike | None = None) -> Array:
+        preferred_element_type: DTypeLike | None = None,
```

Comment:
What's the use case for using `jax.lax.dot`?  I am not against adding this BTW.

Can you also add a test?

Reply:
> I am just testing the interfaces in the lax layer.

What does this mean?

Reply:
That doesn't seem like an explanation :)

Maybe you should exclude this API from your tests. I want to add out_sharding to APIs if it'll be used.

Reply:
@yashk2810 

jax.lax.dot is a public API. Are you not going to support it?  If not, please clearly spficiy it in the document.

Reply:
I have no problems supporting it if you provide a use case! Best to file an issue and I can take a look.